initfiles/componentfiles/thor/run_thor

```diff
@@ -69,41 +69,65 @@ while [ 1 ]; do
     nohup $instancedir/thormaster_$THORNAME MASTER=$THORMASTER:$THORMASTERPORT 2> /dev/null 1>/dev/null &
 
     thorpid=$!
-    if [ "$thorpid" -ne "0" ]; then 
+    if [[ "$thorpid" -ne "0" ]]; then
         echo thormaster$LCR process started pid = $thorpid
         
         echo $thorpid > $PID_NAME
         echo $thorpid > $PID_NAME_BASE
         wait $thorpid
         errcode=$?
-        if [ "$errcode" -eq "0" ]; then  # allows for CTRL-C or thorstop
-            echo thor exited cleanly - script stopping
-            exit 0
-        fi
-        echo master exited with errorcode = $errcode
-        if [ ! -e $SENTINEL ]; then
-            echo $SENTINEL 'has been removed (1) - script stopping'
+        case $errcode in
+        # TEC_Clean
+        0)  echo "Thormaster ($thorpid) exited cleanly"
             $deploydir/stop_thor $deploydir
             exit 0
-        fi
-
-        echo 'stopping thor(slaves) for restart'
-        $deploydir/stop_thor $deploydir keep_sentinel
-
-        if [ 0 != $autoSwapNode ]; then
-            echo "Running autoswap $THORNAME"
-            compname=`basename $PWD`
-            swapnode auto $DALISERVER $compname
-            errcode=$?
-            if [ 0 != ${errcode} ]; then
-                echo auto swap node failed, errcode=${errcode}
+            ;;
+        # TEC_CtrlC
+        1)  echo "Thormaster ($thorpid) Interrupted, Ctrl-C caught"
+            $deploydir/stop_thor $deploydir
+            exit 1
```

Comment:
stop_thor uses the kill_process function from initfiles/sbin/hpcc_setenv.in.  I use it now for all the components so the binaries shut down in the same manner.  SIGTERM is what is being sent to the thormaster process.  Which is why HPCC-13156 exists.  Right now we're always seeing errors in the thormaster logs regarding shutdown.